File: baekjoon/p01598/Main.java
package baekjoon.p01598;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        System.out.println(solve(a, b));
    }

    private static int solve(int a, int b) {
        int ai = (a - 1);
        int ax = (ai / 4);
        int ay = (ai % 4);

        int bi = (b - 1);
        int bx = (bi / 4);
        int by = (bi % 4);

        int x = Math.abs(ax - bx);
        int y = Math.abs(ay - by);

        return (x + y);
    }
}

File: baekjoon/p01598/Main.kt
package baekjoon.p01598

import kotlin.math.abs

fun main() {
    val (a, b) = readLine()!!.split(" ").map { it.toInt() }
    println(solve(a, b))
}

private fun solve(a: Int, b: Int): Int {
    val ai = (a - 1)
    val ax = ai / 4
    val ay = ai % 4

    val bi = (b - 1)
    val bx = bi / 4
    val by = bi % 4

    val x = abs(ax - bx)
    val y = abs(ay - by)

    return (x + y)
}